src/worldcereal/openeo/feature_extractor.py

```diff
@@ -144,6 +146,41 @@ def _rolling_fill(darr, max_iter=2):
 
             return _rolling_fill(darr, max_iter=max_iter)
 
+        def _downsample(arr, factor):
+            """
+            Downsamples a 2D NumPy array by a given factor with average resampling and reflect padding.
+
+            Parameters:
```

Comment:
Let's keep the same docstyle by adding the "-------" under `Parameters`

Reply:
Same for `Returns:`

Reply:
argh didn't use the automatic docstring formatting. Will fix this.